src/sentry/utils/snuba.py

```diff
@@ -904,18 +904,39 @@ def bulk_snuba_queries(
     query_source: (
         QuerySource | None
     ) = None,  # TODO: @athena Make this field required after updated all the callsites
+) -> ResultSet:
+    """
+    Alias for `concurrent_snuba_queries` to use the same referrer for every request.
+    """
+
+    metrics.incr("snql.sdk.api", tags={"referrer": referrer or "unknown"})
+
+    return concurrent_snuba_queries(
+        [(request, referrer) for request in requests],
+        use_cache=use_cache,
+        query_source=query_source,
+    )
+
+
+def concurrent_snuba_queries(
```

Comment:
🤔 should we just have this fully replace `bulk_snuba_queries`? call the old one _deprecated or something

Reply:
It might be nice to keep `bulk_snuba_queries` around for compatibility and convenience unless you'd like to pass a referrer for each query you pass. I'm okay with getting rid of it too but it feels very verbose in most use cases.

Reply:
🤔 I think we could replace the old function with like (needs a better name lmao) `bulk_snuba_queries_with_same_referrer` that just calls this current function with
```python
concurrent_snuba_queries([(request, referrer) for request in requests], use_cache, query_source)
```

Reply:
That's what the old function is doing now. So you're proposing we rename both functions to make it clear which one uses 1 referrer and which one uses multiple referrers?

Reply:
d'oh missed that, actually the more i think about it changing the name of the existing one will suck cause you have to change all the references

so maybe just rename the current function to `bulk_snuba_queries_with_referrers`? 🤔
I'm just worried with `concurrent_snuba_queries` and `bulk_snuba_queries` at a glance its not obvious which one to use

Reply:
Yep, that's precisely my concern. I can definitely rename the second one to `bulk_snuba_queries_with_referrers` if that makes the intent clear that it is to be used when you want to make parallel queries with different referrers.

Reply:
yeah i feel like longterm we need to drop the other one, its easy enough to write the `with_same_referrer` version yourself.

Reply:
Sounds good, gonna rename it to `bulk_snuba_queries_with_referrers` to start and we can think about migrating to it everywhere in the future.

Reply:
~do it in this PR!~
:+1: sounds good to me